server/odc-service/src/main/java/com/oceanbase/odc/service/connection/ConnectionValidator.java

```diff
@@ -50,6 +50,12 @@ void validateForUpsert(ConnectionConfig connection) {
         PreConditions.validInHostWhiteList(connection.getHost(), connectProperties.getHostWhiteList());
         PreConditions.validArgumentState(environmentService.exists(connection.getEnvironmentId()),
                 ErrorCodes.BadRequest, null, "invalid environment id");
+        if (connection.getType().isDefaultSchemaRequired()) {
+            PreConditions.notBlank(connection.getDefaultSchema(), "connection.defaultSchema");
+        }
+        if (connection.getDialectType().isOracle()) {
+            connection.setDefaultSchema(null);
```

Comment:
skip defaultSchema for oracle

Reply:
It seems like not useful, we will set default schema when creating session:

```java
ConnectSessionService#create

@SkipAuthorize("check permission internally")
    public ConnectionSession create(@NotNull CreateSessionReq req) {
        ...
        if (StringUtils.isNotEmpty(schemaName)) {
            connection.setDefaultSchema(schemaName);
        }
        ...
    }
```

Reply:
this pr #1797 involved this problem. let us check this problem in detail:
1.  defaultSchema should not be set by the front end at any time except for odp-sharding
2. we will ignore the default schema when connect type is odp-sharding if we add `JsonIgnore` at the filed which cause this issue

so, can we do this to fix this issue:
1. remove the `JsonIgnore` annotation from the `defaultschema` field
2. make odp-sharding as a special case, that is to say we should do this in two method `ConnectionService#innerCreate` and `ConnectionService#update`, like this:
```java
if (connection.getDialectType != DialectType.ODP_SHARDING) {
    connection.setDefaultSchema(null);
}
```

Reply:
`setDefaultSchema(null)` move to ConnectionService now